daffodil-runtime1/src/main/scala/org/apache/daffodil/dsom/CompiledExpression1.scala

```diff
@@ -200,14 +205,23 @@ class DPathCompileInfo(
   val namespaces: scala.xml.NamespaceBinding,
   val path: String,
   override val schemaFileLocation: SchemaFileLocation,
-  val tunable: DaffodilTunables)
+  val tunable: DaffodilTunables,
+  @TransientParam val typeCalcMapArg: TypeCalcMap)
```

Comment:
Does this variable need to be passed by-name? I forget why we had to do that, but all other TransientParams are by-name args.

Reply:
No. All passed-by-name args need to be transient; and that is normally the only reason we need to use TransientParam.

In this case, we are using TransientParam because (for an unknown reason) it is simply to possible to unserialize the value, so we avoid doing so. Like with the by-name parameters, we pull a similar trick of having a corresponding field which is serializable.

This is the same reason why the corresponding typeCalcMap field is not marked as lazy.

Reply:
I would have made the same observation as Steve. So this needs to go into the code as commentary so we understand that this transient param is different from others.

Reply:
On second thought, there is no reason to not make this passed by-name. Even though the TransientParam is there for a different reason, spamming lazyness is still our strategy to avoid circular dependencies, and it can't hurt (much) to do so preemptivly.